node/src/components/linear_chain/state.rs

```diff
@@ -2,37 +2,48 @@ use std::collections::HashMap;
 
 use datasize::DataSize;
 use itertools::Itertools;
+use num::rational::Ratio;
 use tracing::{debug, warn};
 
 use casper_hashing::Digest;
-use casper_types::{ExecutionResult, ProtocolVersion};
+use casper_types::{EraId, ExecutionResult, ProtocolVersion};
 
 use super::{
     pending_signatures::PendingSignatures, signature::Signature, signature_cache::SignatureCache,
 };
-use crate::types::{Block, BlockHash, BlockSignatures, DeployHash, FinalitySignature};
+use crate::{
+    components::{consensus, linear_chain_sync::KeyBlockInfo},
+    types::{ActivationPoint, Block, BlockHash, BlockSignatures, DeployHash, FinalitySignature},
+};
 
 #[derive(DataSize, Debug)]
 pub(crate) struct LinearChain {
     /// The most recently added block.
     latest_block: Option<Block>,
+    /// The key block information of the most recent eras.
+    key_block_info: HashMap<EraId, KeyBlockInfo>,
     /// Finality signatures to be inserted in a block once it is available.
     pending_finality_signatures: PendingSignatures,
     signature_cache: SignatureCache,
     /// Current protocol version of the network.
     protocol_version: ProtocolVersion,
     auction_delay: u64,
     unbonding_delay: u64,
+    /// The fraction of validators, by weight, that have to sign a block to prove its finality.
+    #[data_size(skip)]
```

Comment:
Why the `skip` here? Is that because `Ratio` doesn't implement `DataSize`?

Reply:
Yes, exactly. Also, it's negligible.